perfmetrics/scripts/bigquery/bigquery.py

```diff
@@ -0,0 +1,206 @@
+import argparse
+from google.cloud import bigquery
+import sys
+
+PROJECT_ID = 'gcs-fuse-test'
+DATASET_ID = 'performance_metrics'
+CONFIGURATION_TABLE_ID = 'experiment_configuration'
+FIO_TABLE_ID = 'fio_metrics'
+VM_TABLE_ID = 'vm_metrics'
+LS_TABLE_ID = 'ls_metrics'
+
+# Construct a BigQuery client object.
+client = bigquery.Client()
+
+class BigQuery():
+
+  def insert_config_and_get_config_id(self, gcsfuse_flags, branch, end_date) -> int:
+
+    """Gets the experiment configuration and checks if it is already present in the BigQuery tables,
+       If not present: New configuration will be inserted, and it's configuration id will be returned
+       If present: Configuration id will be fetched and returned
+
+    Args:
+      gcsfuse_flags: GCSFuse flags for mounting the test buckets.
+      branch: GCSFuse repo branch to be used for building GCSFuse.
+      end_date: Date upto when tests are run.
+
+    Return:
+      Configuration id of the experiment in the BigQuery tables
+    """
+    dataset_ref = None
+    try:
+      dataset_ref = client.dataset(DATASET_ID)
+    except:
+      dataset_ref = client.create_dataset(DATASET_ID)
+
+    # Query for creating experiment_configuration table if it does not exist
+    query_create_table_experiment_configuration = """
+        CREATE TABLE IF NOT EXISTS {}.{}.{}(
+          configuration_id INT64,
+          gcsfuse_flags STRING,
+          branch STRING,
+          end_date TIMESTAMP,
+          PRIMARY KEY (configuration_id) NOT ENFORCED
+        ) OPTIONS (description = 'Table for storing Job Configurations and respective VM instance name on which the job was run');
+    """.format(PROJECT_ID, DATASET_ID, CONFIGURATION_TABLE_ID)
+
+    # API Request to create experiment_configuration if it does not exist
+    results = client.query(query_create_table_experiment_configuration)
```

Comment:
What's the value of result in both cases - if table exists and if not exist?

Reply:
It returns a 'google.cloud.bigquery.job.QueryJob' object. This object represents that query job that was executed in BigQuery. The result will look something like 'QueryJob<project=gcsfuse-intern-project-2023, location=US, id=35253b73-68da-4d48-b19d-b922609752b9>'. We can remove statements to print the result. This query cannot result in error as the table will only be created if it does not exist.